### DISCUSSION TITLE
How can I persist db data?

### DISCUSSION THREAD

**Discussion by 0xSalman** [dragonflydb/dragonfly#1050] [Q&A] [answered]
Hello,

How can I persist the db data? I am running the binary via a systemd service on my linux server. I have noticed that data is not persisting on restarts. 

I added a simple key via `set hello world` and after I restarted the service, the key was gone. 

```
/usr/bin/dragonfly \
  --bind=localhost \
  --port=7379 \
  --dir=/var/lib/dragonfly/dev \
  --cache_mode=true \
  --pidfile=/var/run/dragonfly/dev.pid \
  --log_dir=/var/log/dragonfly/dev \
  --dbfilename=dump.rdb \
  --maxmemory=2gb \
  --save_schedule="*:05" \
  --version_check=false
```

I thought `save_schedule` would persist the db file but it seems it only saves a snapshot. I would appreciate any help, thank you!

**Comment by heikomat** [comment, marked as answer]
You have one of two options
1. remove `--dbfilename=dump.rdb`
   This makes dragonfly load the data from the latest snapshot instead of trying to read it from `dump.rdb`
2. add `--df_snapshot_format=false`
   This makes dragonfly save snapshots in the `dump.rdb`, instead of its own snapshotting format

Either should result in a snapshot/dump being created every 5 minutes (with your config) that should get loaded on restart
